### PR TITLE
new option for disable network scan

### PR DESCRIPTION
**Pull request by ctapmex**
## Summary
New option for disable network scan.

 
## References
https://forum.farmanager.com/viewtopic.php?p=169115#p169115

## Checklist
- [x] I have followed the [contributing guidelines](https://github.com/FarGroup/FarManager/blob/master/CONTRIBUTING.md).
- [ ] I have discussed this with project maintainers:  <br/>
If not checked, I accept that this work might be rejected in favor of a different grand plan.<br/>

### CODE REVIEW

File: plugins/network/NetEng.hlf [hunk 1]
```diff
@@ -44,30 +44,34 @@ $ #Network browser configuration#
  Enables adding "Network" item to the disks menu.
 
  #Add to plugins menu#
- Enables adding "Network" item to the plugins menu.
+ Enables adding "Network browser" item to the plugins menu.
 
- #Выводить ".." в списке провайдеров#
- Показывать ".." на панели в списке провайдеров.
+ #Show ".." in the providers#
```
Comment (alabuzhev) [inline plugins/network/NetEng.hlf:49]: `the providers list` or `the list of providers`

File: plugins/network/NetEng.hlf [hunk 2]
```diff
@@ -44,30 +44,34 @@ $ #Network browser configuration#
...
 workgroup). The option works only for Microsoft Windows networks.
 
- #Показывать принтеры#
- Показывать предоставляемые хостом общие принтеры.
+ #Show printers#
+ Show shared printers.
 
- #Показывать полные пути к общим ресурсам#
- ...
+ #Show share full name#
```
Comment (alabuzhev) [inline plugins/network/NetEng.hlf:60]: Probably `full share name`

File: plugins/network/NetEng.hlf [hunk 3]
```diff
@@ -44,30 +44,34 @@ $ #Network browser configuration#
...
+ When the option is enabled, \\Server1 and \\Server1\Share1 are displayed, when disabled - Server1 and Share1
 
  #Show hidden shares#
- If this option is enabled, hidden (administrative) shares are shown in the plugin.
+ Опция управляет показом скрытых (административных) общих ресурсов компьютера.
```
Comment (alabuzhev) [inline plugins/network/NetEng.hlf:64]: ?

File: plugins/network/NetEng.hlf [hunk 4]
```diff
@@ -44,30 +44,34 @@ $ #Network browser configuration#
...
 
- #Показывать скрытые ресурсы атрибутом Hidden#
- Если включена эта опция, плагин для скрытых административных ресурсов ставит атрибут Hidden.
+ - #Never show#
+   There is no search for hidden shares.
```
Comment (alabuzhev) [inline plugins/network/NetEng.hlf:67]: *"There is no spoon"* :)
`Don't show hidden shares` or `hidden shares are never shown` or something like that might sound more natural.

File: plugins/network/NetEng.hlf [hunk 5]
```diff
@@ -44,30 +44,34 @@ $ #Network browser configuration#
  Enables adding "Network" item to the disks menu.
 
  #Add to plugins menu#
- Enables adding "Network" item to the plugins menu.
+ Enables adding "Network browser" item to the plugins menu.
```
Comment (alabuzhev) [inline plugins/network/NetEng.hlf:47]: The English hlf is the baseline for all the other translations.
It's easier to spot the untranslated bits by comparing with it.
If you change anything in it, please update all the other hlfs as well, it's a pain when they go out of sync.

File: plugins/network/NetCfg.cpp [hunk 6]
```diff
@@ -52,16 +53,17 @@ int Config()
...
+			Opt.HiddenSharesAsHidden = false;
 			break;
 		case 2: // always show
-			Opt.HiddenShares = TRUE;
-			Opt.HiddenSharesAsHidden = FALSE;
+			Opt.HiddenShares = true;
+			Opt.HiddenSharesAsHidden = false;
+			break;
+		case 1: // make hidden
+		default:
```
Comment (alabuzhev) [inline plugins/network/NetCfg.cpp:64]: Why do we need this change?
The value comes from a dialog, we won't suddenly get `42` here.

Comment (ctapmex) [reply plugins/network/NetCfg.cpp:64]: >we won't suddenly get 42 here.
sure ? ;) ok, even if this happens, then the cost of error here is small

File: plugins/network/NetEng.hlf (hunk 1 above)
Comment (ctapmex) [reply plugins/network/NetEng.hlf:49]: I took these lines (and below) from the NetEng.lng. But ok, once we started, let's put it in order.

File: plugins/network/NetEng.hlf (hunk 5 above)
Comment (BestiaPL) [reply plugins/network/NetEng.hlf:47]: Please add the "Scan network" topic to help file - now it is not described.

Comment (ctapmex) [reply plugins/network/NetEng.hlf:47]: @BestiaPL  in the next patch

**Pull request merged**